e2e/framework/framework.go

```diff
@@ -140,11 +141,17 @@ func InitializeCluster() {
 			Expect(err).NotTo(HaveOccurred())
 
 			withInstaller := TestContext.Onprem.InstallerURL != "" && TestContext.Wizard
-			if TestContext.Provisioner.LoadFromState {
-				installerNode, err = provisioner.LoadFromState(TestContext.Provisioner.StateFile, withInstaller)
+			loader, ok := provisioner.(infra.ExternalStateLoader)
```

Comment:
Not directly related to this PR but it's something I'd put in in the beginning to avoid having a `trace.NotImplemented` for `vagrant` as the ability to load state from external state file is only relevant for `terraform` provisioner (i.e. `terraform.tfstate`)